MuckRock/documentcloud-frontend
# Document Revisions UI

This update adds UI for enabling and viewing a document's revision history. This feature is gated to premium accounts, with upgrade touts for free users.

Revision control can be enabled when uploading documents, or after from the viewer. The document's revision history can be accessed from the viewer.

Note: Some stories are not rendering correctly in Chromatic. I'm opening a new branch to figure out why. As this is only an internal issue to our deployment flow, I'm ignoring these stories in Chromatic for the time being.

### Testing

1. Log in to deploy preview.
2. As a free user, upload a document.
    1. The revision history option should be disabled, with a link to upgrade your account.
    2. After uploading, view the document. The sidebar action for Revision History should be clickable, but present an upgrade tout.
3. As a premium user, upload a document.
    1. The revision history option should be selectable. Enable revision history for your upload.
    2. After upload completes, view the document. The sidebar action for Revision History should be clickable, and present the initial version and OCR'd version of the document.
    3. Make a change, like redacting a portion of the document. After it completes, the revision history should show a new entry.
    4. Try downloading a previous version of the document. It should not include any subsequent changes made.

### Screenshots

![image](https://github.com/MuckRock/documentcloud-frontend/assets/6089482/f78c938d-6e7c-43bb-87a3-06d468b0ac2f)


![image](https://github.com/MuckRock/documentcloud-frontend/assets/6089482/36add03a-3954-4e62-9d67-bcf73084555f)
